src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -231,6 +231,15 @@ class PostTyper extends MacroTransform with IdentityDenotTransformer  { thisTran
                   ctx.compilationUnit.source.exists &&
                   sym != defn.SourceFileAnnot)
                 sym.addAnnotation(Annotation.makeSourceFile(ctx.compilationUnit.source.file.path))
+
+              for (parent <- sym.asClass.classInfo.classParents) {
+                val pclazz = parent.classSymbol
+                if (pclazz.is(Sealed)) {
+                  println(s"add $sym to $pclazz")
+                  pclazz.addAnnotation(Annotation.makeChild(sym))
+                }
+              }
+
```

Comment:
Anonymous classes caused pickling-related error for `tests/run/t8611b.scala`:

https://gist.github.com/liufengyun/f4aca0853e06f96c9130a0a05bfa0d4f

If I ignore anonymous classes, then it works:

``` Scala
              if (!sym.isAnonymousClass) // ignore anonymous class
                for (parent <- sym.asClass.classInfo.classParents) {
                  val pclazz = parent.classSymbol
                  if (pclazz.is(Sealed)) pclazz.addAnnotation(Annotation.makeChild(sym))
                }
```

The problem is that ignore anonymous classes would make the algorithm silent in case it should report a warning:

``` Scala
sealed trait A
case class B() extends A
case class C() extends A

object M {
  val x = new A {}

  x match {
     case _: B => true
     case _: C => true
  }
}
```

Any ideas, @odersky @smarter ?